code/obj/machinery/food_and_drink/mixer.dm

```diff
@@ -239,13 +241,27 @@ TYPEINFO(/obj/machinery/mixer)
 		UnsubscribeProcess()
 		return
 
+	power_change()
+		. = ..()
+		src.UpdateIcon()
+
 	update_icon()
 		if (!src || !istype(src))
 			return
 
-		if (src.working != 0)
-			src.icon_state = "blender_on"
+		if (length(src.contents) > 0)
+			src.icon_state = "blender_filled"
+		else
+			src.icon_state = "blender_empty"
+
+		src.overlays.len = 0
+		var/powered = src.status & NOPOWER
+
+		if (powered)
+			src.overlays += image(src.icon, "blender_off")
+		else if (!powered && !src.working)
+			src.overlays += image(src.icon, "blender_powered")
```

Comment:
Cleaner to use OverlayManager and such

Reply:
Did you mean UpdateOverlays by any chance? Struggling to understand how I would use anything OverlayManager to do that.

Reply:
Yes